### PR TITLE
kubevirt: Fix deploying kubevirt on git actions

### LINKED ISSUE #1844: [main] IPAM lane broken
**What happened**:
IPAM lane seems broken on `main` branch: see [lane failure](https://github.com/kubevirt/cluster-network-addons-operator/actions/runs/10250317432/job/28355761470?pr=1842):
```
+ ./cluster/kubectl.sh wait -n kubevirt kv kubevirt --for condition=Available --timeout 15m
error: timed out waiting for the condition on kubevirts/kubevirt
+ echo 'KubeVirt not ready in time'
KubeVirt not ready in time
```

**What you expected to happen**:
lane to pass

**How to reproduce it (as minimally and precisely as possible)**:
Steps to reproduce the behavior.
run the lane

**Additional context**:
This issue is currently blocking https://github.com/kubevirt/cluster-network-addons-operator/pull/1841

**Environment**:
- KubeVirt version (use `virtctl version`): `v1.3.0`

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Since we are using git actions, the resources are limited.

We saw out of space errors, fix them by cleaning the container before
deploying the pods.

Also don't deploy some components to free more memory resources.
    
**Special notes for your reviewer**:
Fixes #1844 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/cc @RamLavi 
lets try this approach as well

**Comment by oshoval** [comment]
nope didnt work

**Comment by oshoval** [comment]
ok seems it is the wait bug, that need to be split

File: hack/deploy-kubevirt.sh [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 set -exo pipefail
 
+if [ "$GITHUB_ACTIONS" == "true" ]; then
+    for node in $(./cluster/kubectl.sh get node --no-headers  -o custom-columns=":metadata.name"); do
+        docker exec -t $node bash -c "echo 'fs.inotify.max_user_watches=1048576' >> /etc/sysctl.conf"
+        docker exec -t $node bash -c "echo 'fs.inotify.max_user_instances=512' >> /etc/sysctl.conf"
+        docker exec -i $node bash -c "sysctl -p /etc/sysctl.conf"
```
Comment (maiqueb) [inline hack/deploy-kubevirt.sh:8]: can we please stop writing code that only works on docker ? 

does this happen elsewhere in this project ?

**Review by maiqueb — CHANGES_REQUESTED** [review]

File: hack/deploy-kubevirt.sh (hunk 1 above)
Comment (oshoval) [reply hack/deploy-kubevirt.sh:8]: it is WIP, please wait for the review

Comment (oshoval) [reply hack/deploy-kubevirt.sh]: ok seems the problem is that the first one out of CNAO / kubevirt is deployed fine, but the 2nd one (tried both orders) isnt
i can add logs to make sure it is memory, but it looks like that

we might want to check if OVN use self-hosted runners
and we can consider moving to prow imo at least as first step

**Comment by oshoval** [comment]
```
Unhandled exception. System.IO.IOException: No space left on device : '/home/runner/runners/2.317.0/_diag/Worker_20240806-084825-utc.log'
   at System.IO.RandomAccess.WriteAtOffset(SafeFileHandle handle, ReadOnlySpan`1 buffer, Int64 fileOffset)
   at System.IO.Strategies.BufferedFileStreamStrategy.FlushWrite()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
System.IO.IOException: No space left on device : '/home/runner/runners/2.317.0/_diag/Worker_20240806-084825-utc.log'
   at System.Diagnostics.TextWriterTraceListener.Flush()
   at System.IO.RandomAccess.WriteAtOffset(SafeFileHandle handle, ReadOnlySpan`1 buffer, Int64 fileOffset)
   at System.IO.Strategies.BufferedFileStreamStrategy.FlushWrite()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.Diagnostics.TextWriterTraceListener.Flush()
   at GitHub.Runner.Common.HostTraceListener.WriteHeader(String source, TraceEventType eventType, Int32 id)
   at GitHub.Runner.Common.HostTraceListener.TraceEvent(TraceEventCache eventCache, String source, TraceEventType eventType, Int32 id, String message)
   at System.Diagnostics.TraceSource.Flush()
   at System.Diagnostics.TraceSource.TraceEvent(TraceEventType eventType, Int32 id, String message)
   at GitHub.Runner.Common.TraceManager.Dispose(Boolean disposing)
   at GitHub.Runner.Common.TraceManager.Dispose()
   at GitHub.Runner.Worker.Worker.RunAsync(String pipeIn, String pipeOut)
   at GitHub.Runner.Common.HostContext.Dispose(Boolean disposing)
   at GitHub.Runner.Common.HostContext.Dispose()
   at GitHub.Runner.Worker.Program.MainAsync(IHostContext context, String[] args)
   at GitHub.Runner.Worker.Program.Main(String[] args)
System.IO.IOException: No space left on device : '/home/runner/runners/2.317.0/_diag/Worker_20240806-084825-utc.log'
   at System.IO.RandomAccess.WriteAtOffset(SafeFileHandle handle, ReadOnlySpan`1 buffer, Int64 fileOffset)
   at System.IO.Strategies.BufferedFileStreamStrategy.FlushWrite()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.Diagnostics.TextWriterTraceListener.Flush()
   at GitHub.Runner.Common.HostTraceListener.WriteHeader(String source, TraceEventType eventType, Int32 id)
   at GitHub.Runner.Common.HostTraceListener.TraceEvent(TraceEventCache eventCache, String source, TraceEventType eventType, Int32 id, String message)
   at System.Diagnostics.TraceSource.TraceEvent(TraceEventType eventType, Int32 id, String message)
   at GitHub.Runner.Common.Tracing.Error(Exception exception)
   at GitHub.Runner.Worker.Program.MainAsync(IHostContext context, String[] args)
```

yuck
i think we should move to prow

**Comment by oshoval** [comment]
https://github.com/kubevirt/cluster-network-addons-operator/actions/runs/10263335275/job/28395542383?pr=1843
again storage problem, even that this PR changes are naive as far as i can tell
this is a new problem that started just on the last 2 runs

```
crypto/subtle: write /tmp/go-build334935846/b070/importcfg: no space left on device
crypto/internal/boring/sig: mkdir /tmp/go-build334935846/b072/: no space left on device
```

**Comment by oshoval** [comment]
Going to try to use kind-ovn kubevirtci provider that already worked on prow when was tested with kubevirt ipam PR
as the errors here arent recoverable easily or at all (limited resources on git actions for heavy testing requirement, as we deploy lots of pods)

**Comment by oshoval** [comment]
awesome passed now
https://github.com/kubevirt/cluster-network-addons-operator/actions/runs/10264945110/job/28400069861?pr=1843
thanks for the co-op Miguel

will rework the PR

**Comment by oshoval** [comment]
ready for review please
thanks

File: automation/components-functests.setup.sh [hunk 2]
```diff
@@ -66,16 +66,16 @@ spec:
...
   kubevirtIpamController: {}
   imagePullPolicy: Always
 EOF
 
   if [[ $USE_KUBEVIRTCI == true ]]; then
     echo "  multus: {}" >> cr.yaml
     echo "  multusDynamicNetworks: {}" >> cr.yaml
+    echo "  ovs: {}" >> cr.yaml
+    echo "  macvtap: {}" >> cr.yaml
+    echo "  kubeSecondaryDNS: {}" >> cr.yaml
```
Comment (maiqueb) [inline automation/components-functests.setup.sh:78]: why do we need this ?

**Review by maiqueb — CHANGES_REQUESTED** [review]
Why do we need linux bridge plugin ?

File: automation/components-functests.setup.sh (hunk 2 above)
Comment (oshoval) [reply automation/components-functests.setup.sh:78]: the fact it is here means dont deploy it on our lane
before this PR it deployed those, now it doesnt
so it free memory

Comment (maiqueb) [reply automation/components-functests.setup.sh:78]: ok. Let's at least remove the linux-bridge. 

I wonder if we can fly without kubemacpool.

... but we can understand that later on.

Comment (oshoval) [reply automation/components-functests.setup.sh:78]: done

**Review by maiqueb — APPROVED** [review]
Thank you.

**Comment by oshoval** [comment]
/retest 

not related

**Pull request merged**